backends/jms/src/main/java/org/hibernate/search/backend/impl/jms/AbstractJMSHibernateSearchController.java

```diff
@@ -69,8 +70,16 @@ public void onMessage(Message message) {
 				log.messageReceivedForUndefinedIndex( indexName );
 				return;
 			}
-			queue = indexManager.getSerializer().toLuceneWorks( (byte[]) objectMessage.getObject() );
-			indexManager.performOperations( queue, null );
+
+			LuceneWorkSerializer luceneWorkSerializer = integrator.getServiceManager().requestService( LuceneWorkSerializer.class );
```

Comment:
Ok, that sounds great. I'll rework it then, adding back that method to the {{IndexManager}} contract, it would just not be used really.